### PR TITLE
added a document on experiment tracking using kfp

### PR DESCRIPTION
**Pull request by oindrillac**
related #72

### CODE REVIEW

File: docs/track-metrics-using-kubeflow.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ 
+Kubeflow pipelines also allow us to [export metrics](https://www.kubeflow.org/docs/components/pipelines/sdk/pipelines-metrics/) from a running pipeline which can help compare multiple runs for example over multiple model training runs. We can compare the achieved model performance metrics like accuracy, f1 score achieved.
+
+## Pre-requisites
+
+1. Existing kubeflow pipeline with components (notebooks or scripts) for which you want to track metrics.
+
+## Track Metrics
+
+To enable tracking of metrics during the execution of a machine learning experiment, your component must have an output called `mlpipeline-metrics`  which must return a JSON-serialized metrics dictionary. Let's demonstrate this with the help of this [notebook](https://github.com/os-climate/aicoe-osc-demo/blob/master/notebooks/demo1/demo1-create-tables.ipynb).
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:15]: Can we remove the link to that notebook here?  Move it to the very bottom of the doc, add something like, "if you want to see an real example of how to integrate experiement tracking, take a look at this noteboook"

File: docs/track-metrics-using-kubeflow.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Experiment tracking using Kubeflow Pipelines
+
+In this guide you will see how you can use kubeflow pipelines to track metrics from different experiments and runs.
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:3]:
```suggestion
In this guide you will learn how to use kubeflow pipelines to track metrics from different experiments and runs.
```

File: docs/track-metrics-using-kubeflow.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+[Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/) is a platform for building and deploying scalable machine learning (ML) workflows. It allows us to automate the running of the jupyter notebooks and scripts using a simple workflow. Using Kubeflow pipelines we can automate the running of multiple steps of a ML pipeline like data collection, model training, inference, which need to be performed sequentially or parallel.
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:5]:
```suggestion
[Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/) is a platform for building and deploying scalable machine learning (ML) workflows. It allows us to automate the running of  jupyter notebooks and scripts using a simple workflow. Using Kubeflow pipelines we can automate the running of multiple steps of a ML pipeline like data collection, model training, inference, and control which steps need to be performed sequentially and which should run in  parallel.
```

File: docs/track-metrics-using-kubeflow.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ 
+Kubeflow pipelines also allow us to [export metrics](https://www.kubeflow.org/docs/components/pipelines/sdk/pipelines-metrics/) from a running pipeline which can help compare multiple runs for example over multiple model training runs. We can compare the achieved model performance metrics like accuracy, f1 score achieved.
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:7]:
```suggestion
Kubeflow pipelines also allow us to [export metrics](https://www.kubeflow.org/docs/components/pipelines/sdk/pipelines-metrics/) from a running pipeline. This helps us compare multiple runs over multiple model training runs. We can also compare the achieved model performance metrics like accuracy, f1, etc.
```

File: docs/track-metrics-using-kubeflow.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+## Pre-requisites
+
+1. Existing kubeflow pipeline with components (notebooks or scripts) for which you want to track metrics.
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:11]:
```suggestion
1. Existing kubeflow pipeline instance available and the components (notebooks or scripts) that you want to track metrics for.
```

File: docs/track-metrics-using-kubeflow.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+## Track Metrics
+
+To enable tracking of metrics during the execution of a machine learning experiment, your component must have an output called `mlpipeline-metrics`  which must return a JSON-serialized metrics dictionary. Let's demonstrate this with the help of this [notebook](https://github.com/os-climate/aicoe-osc-demo/blob/master/notebooks/demo1/demo1-create-tables.ipynb).
+
+In this notebook, we want to capture metrics such as the time taken to execute certain queries, and the time taken to upload some files to S3 storage.
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:17]:
```suggestion
In this notebook, we want to capture metrics such as the time taken to execute certain queries, and the time taken to upload some files to S3 storage.
```
```suggestion
Below we will demonstrate how to capture metrics for the time taken to execute certain queries, and the time taken to upload some files to S3 storage.
```

File: docs/track-metrics-using-kubeflow.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Experiment tracking using Kubeflow Pipelines
```
Comment (MichaelClifford) [inline docs/track-metrics-using-kubeflow.md:1]: Is there a video link we can embed here also?

**Review by MichaelClifford — CHANGES_REQUESTED** [review]
:+1:

**Review by MichaelClifford — CHANGES_REQUESTED** [review]
:+1:

File: docs/track-metrics-using-kubeflow.md (hunk 1 above)
Comment (oindrillac) [reply docs/track-metrics-using-kubeflow.md:15]: sounds good.

File: docs/track-metrics-using-kubeflow.md (hunk 7 above)
Comment (oindrillac) [reply docs/track-metrics-using-kubeflow.md:1]: video is WIP, can update the PR with the link once ready :)

Comment (oindrillac) [reply docs/track-metrics-using-kubeflow.md:1]: embedded the unlisted video. Please take a look

**Comment by oindrillac** [comment]
@MichaelClifford made the changes and added the video :)

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**